### DISCUSSION TITLE
启动失败，但test却正常

### DISCUSSION THREAD

**Discussion by YipSzeho** [Tsuk1ko/CQ-picfinder-robot#242] [问题求助] [answered]
用npm run test可以正常运行，start就报错

> PS C:\Users\Admin\cq-picsearcher-bot> npm run test

cq-picsearcher-bot@2.30.0 test C:\Users\Admin\cq-picsearcher-bot
node index.js

2021/8/27 下午6:18:27 配置已加载
2021/8/27 下午6:18:29 连接中[/event]#1
2021/8/27 下午6:18:29 连接中[/api]#1
2021/8/27 下午6:18:30 连接成功[/event]#1
2021/8/27 下午6:18:30 连接成功[/api]#1
终止批处理操作吗(Y/N)? y

> PS C:\Users\Admin\cq-picsearcher-bot> npm start

cq-picsearcher-bot@2.30.0 start C:\Users\Admin\cq-picsearcher-bot
pm2 start ecosystem.config.js

[PM2] Applying action restartProcessId on app [cqps](ids: [ 0 ])
[PM2][ERROR] Process 0 not found
Process 0 not found
C:\Users\Admin\cq-picsearcher-bot\node_modules\pm2\lib\API.js:1722
        acted.push(proc.pm2_env ? proc.pm2_env.pm_id : proc.pm_id)
                        ^

TypeError: Cannot read property 'pm2_env' of undefined
    at C:\Users\Admin\cq-picsearcher-bot\node_modules\pm2\lib\API.js:1722:25
    at Array.forEach (<anonymous>)
    at API.speedList (C:\Users\Admin\cq-picsearcher-bot\node_modules\pm2\lib\API.js:1721:18)
    at C:\Users\Admin\cq-picsearcher-bot\node_modules\pm2\lib\binaries\CLI.js:301:15
    at wrapper (C:\Users\Admin\cq-picsearcher-bot\node_modules\async\internal\once.js:12:16)
    at replenish (C:\Users\Admin\cq-picsearcher-bot\node_modules\async\internal\eachOfLimit.js:76:25)
    at iterateeCallback (C:\Users\Admin\cq-picsearcher-bot\node_modules\async\internal\eachOfLimit.js:65:17)
    at C:\Users\Admin\cq-picsearcher-bot\node_modules\async\internal\onlyOnce.js:12:16
    at C:\Users\Admin\cq-picsearcher-bot\node_modules\pm2\lib\binaries\CLI.js:292:11
    at C:\Users\Admin\cq-picsearcher-bot\node_modules\pm2\lib\API.js:335:21
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! cq-picsearcher-bot@2.30.0 start: `pm2 start ecosystem.config.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the cq-picsearcher-bot@2.30.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Admin\npm-cache\_logs\2021-08-27T10_18_43_646Z-debug.log
PS C:\Users\Admin\cq-picsearcher-bot>

**Comment by Tsuk1ko** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/24877906/131154441-35f60da9-ec37-447d-8312-ea604efccb74.png)
